packager/VisibilityChecker.cc

```diff
@@ -192,9 +204,27 @@ class PropagateVisibility final {
                 e.addErrorLine(sym.loc(ctx), "Defined here");
             }
         }
+
+        // If sym is an enum value, it can't be exported directly. Instead, its wrapping enum class must be exported.
+        auto enumClass = packager::getEnumClassForEnumValue(ctx.state, sym);
+        if (enumClass.exists()) {
+            if (auto e = ctx.beginError(loc, core::errors::Packager::InvalidExport)) {
+                std::string enumClassName = enumClass.show(ctx);
+                std::string_view enumClassNameView = std::string_view{enumClassName};
```

Comment:
```suggestion
                auto enumClassName = enumClass.show(ctx);
```

I don't think there's much point trying to use `string_view` here. The usages inside string formatting methods will all take the string by reference anyways, and they're already going to allocate a new string when constructing the newly-formatted string, so it doesn't matter if you `move(enumClassName)` in—it's going to copy and allocate a new string to prefix it with `export` anyways.

With that in mind, you can simplify a lot of this code by just operating on `enumClassName`.